src/power_grid_model_io/data_stores/excel_file_store.py

```diff
@@ -197,6 +223,35 @@ def _check_duplicate_values(self, sheet_name: str, data: pd.DataFrame) -> Dict[i
 
         return to_rename
 
+    def _process_uuid_columns(self, data: pd.DataFrame, sheet_name: str) -> pd.DataFrame:
+        first_level = data.columns.get_level_values(0)
+        guid_columns = first_level[first_level.str.endswith("GUID")]
+
+        for guid_column in guid_columns:
+            nr = VISION_EXCEL_LAN_DICT[self._languange][DICT_KEY_NUMBER]
+            add_guid_values_to_cvtr(data, guid_column, self._uuid_cvtr)
+            new_column_name = guid_column.replace("GUID", nr)
+            if guid_column == "GUID":
+                if sheet_name in special_nodes_en:
```

Comment:
is this criterion strict enough? what are the odds that there are duplicate values? If i look at the code, the mapping is quite arbitrary but maybe this is vision's own hack to make it work?

```python
special_nodes_en = [
    "Transformer loads",
    "Sources",
    "Synchronous generators",
    "Wind turbines",
    "Loads",
    "Capacitors",
    "Reactors",
    "Zigzag transformers",
    "Pvs",
]
special_nodes_nl = [
    "Transformatorbelastingen",
    "Netvoedingen",
    "Synchrone generatoren",
    "Windturbines",
    "Belastingen",
    "Condensatoren",
    "Spoelen",
    "Nulpuntstransformatoren",
    "Pv's",
]
```

on the same note: how likely is it that the sheet has a different language compared to the language provided to the excel file store?

Reply:
On this I would have to admit that I do not have enough knowledge. For now this is what I get from Vision 9.7. Not sure what would happen with the next release. Could use a follow-up to polish it if/when we know more.